### PR TITLE
feat: [CI-3367]: Adding warning banner for deprecated images in use

### PR DESCRIPTION
**Pull request by vardanbansal-harness**
##### Summary:

Jira: https://harness.atlassian.net/browse/CI-3367

image

##### Screenshots:

You can use the following comments to re-trigger PR Checks

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier - `fix prettier`

### CODE REVIEW

**Comment by vardanbansal-harness** [comment]
retrigger titlecheck

File: src/modules/70-pipeline/pages/execution/ExecutionLandingPage/ExecutionLandingPage.tsx [hunk 1]
```diff
@@ -227,14 +243,39 @@ export default function ExecutionLandingPage(props: React.PropsWithChildren<unkn
...
+                <FeatureRestrictionBanners
+                  featureNames={[
+                    FeatureIdentifier.ACTIVE_COMMITTERS,
+                    FeatureIdentifier.MAX_BUILDS_PER_MONTH,
+                    FeatureIdentifier.MAX_TOTAL_BUILDS
+                  ]}
+                  module={module}
+                />
+                {deprecatedImages?.length ? (
+                  <Layout.Horizontal
```
Comment (mtran7) [inline src/modules/70-pipeline/pages/execution/ExecutionLandingPage/ExecutionLandingPage.tsx:256]: Can create this as a component / clone FeatureRestrictionBanners to scale for more warnings in the future

File: src/modules/70-pipeline/strings/strings.en.yaml [hunk 2]
```diff
@@ -863,3 +863,5 @@ runPipeline:
   ciCodebaseConfig: CI Codebase Configuration is missing. Codebase is required when the pipeline contains a CI stage that is set to clone codebase.
 buildInfra:
   poolId: Pool Id
+imageVersionDeprecated: Image version outdated
+unsupportedImagesWarning: 'You are using unsupported Harness CI {{summary}}. Build stages might fail, but Harness won’t fail builds automatically.'
```
Comment (mtran7) [inline src/modules/70-pipeline/strings/strings.en.yaml:867]: Would run through w/ @michaelcretzman . What is summary? And it is very hard to read the text `Build stages might fail, but Harness won’t fail builds automatically.` at the end when the embedded text is dynamic. Would consider switching around so the summary is last in the sentence and surrounded by brackets[]?

image

Comment (vardanbansal-harness) [reply src/modules/70-pipeline/strings/strings.en.yaml:867]: He's already reviewed it https://harness.slack.com/archives/C031ZKEFSHG/p1645549072888029?thread_ts=1645508752.435939&cid=C031ZKEFSHG. Asked him again for confirmation though.

File: src/modules/70-pipeline/pages/execution/ExecutionLandingPage/ExecutionLandingPage.tsx (hunk 1 above)
Comment (vardanbansal-harness) [reply src/modules/70-pipeline/pages/execution/ExecutionLandingPage/ExecutionLandingPage.tsx:256]: So I thought of doing the same but this warning message is more of a temporary solution. We may remove this warning going forward.

File: src/modules/70-pipeline/strings/strings.en.yaml (hunk 2 above)
Comment (vardanbansal-harness) [reply src/modules/70-pipeline/strings/strings.en.yaml]: Updated with the finalized suggestion by @michaelcretzman.

File: src/modules/70-pipeline/pages/execution/ExecutionLandingPage/ExecutionLandingPage.tsx (hunk 1 above)
Comment (vardanbansal-harness) [reply src/modules/70-pipeline/pages/execution/ExecutionLandingPage/ExecutionLandingPage.tsx]: Done

File: src/modules/70-pipeline/strings/strings.en.yaml [hunk 3]
```diff
@@ -863,3 +863,5 @@ runPipeline:
   ciCodebaseConfig: CI Codebase Configuration is missing. Codebase is required when the pipeline contains a CI stage that is set to clone codebase.
 buildInfra:
   poolId: Pool Id
+imageVersionDeprecated: Image version outdated
+unsupportedImagesWarning: 'You are using unsupported Harness CI images. Harness allows builds to use these images, but we cannot ensure that builds will not fail. Please update the versions of the following image{{summary}}'
```
Comment (mtran7) [inline src/modules/70-pipeline/strings/strings.en.yaml:867]: Add space after image

Comment (vardanbansal-harness) [reply src/modules/70-pipeline/strings/strings.en.yaml:867]: Done 
image

**Review by mtran7 — APPROVED** [review]

**Review by tan-nhu — APPROVED** [review]

**Pull request merged**